js/scripts.js

```diff
@@ -0,0 +1,131 @@
+var newGameBtn = document.getElementById('js-newGameButton');
+newGameBtn.addEventListener('click', newGame);
+
+var pickRock = document.getElementById('js-playerPick_rock'),
+pickPaper = document.getElementById('js-playerPick_paper'),
+pickScissors = document.getElementById('js-playerPick_scissors');
+
+pickRock.addEventListener('click', function() { playerPick('rock') });
+pickPaper.addEventListener('click', function() { playerPick('paper') });
+pickScissors.addEventListener('click', function() { playerPick('scissors') });
+
+var gameState = 'notStarted',
+player = {
+	name: '',
+	score: 0
+},
+computer = {
+	score: 0
+};
+
+var newGameElem = document.getElementById('js-newGameElement'),
+    pickElem = document.getElementById('js-playerPickElement'),
+    resultsElem = document.getElementById('js-resultsTableElement');
+
+function setGameElements() {
+  switch(gameState) {
+    case 'started':
+        newGameElem.style.display = 'none';
+        pickElem.style.display = 'block';
+        resultsElem.style.display = 'block';
+      break;
+    case 'ended':
+        newGameBtn.innerText = 'Jeszcze raz';
+    case 'notStarted':
+    default:
+        newGameElem.style.display = 'block';
+        pickElem.style.display = 'none';
+        resultsElem.style.display = 'none';
+  }
+}
+
+setGameElements();
+
+var playerPointsElem = document.getElementById('js-playerPoints'),
+    playerNameElem = document.getElementById('js-playerName'),
+    computerPointsElem = document.getElementById('js-computerPoints');
+
+function setGamePoints() {
+    playerPointsElem.innerHTML = player.score;
+    computerPointsElem.innerHTML = computer.score;
+
+    endGame();
+    setGameElements();
+}
+
+function newGame() {
+  player.name = prompt('Graczu, wpisz swoje imię', 'imię gracza');
+  if (player.name) {
+    player.score = computer.score = 0;
+    gameState = 'started';
+    setGameElements();
+
+    playerNameElem.innerHTML = player.name;
+    setGamePoints();
+  }
+
+}
+
+
+var x = Math.random();
+
+function getComputerPick() {
+    var possiblePicks = ['rock', 'paper', 'scissors'];
+    return possiblePicks[Math.floor(Math.random()*3)];
+}
+
+var playerPickElem = document.getElementById('js-playerPick'),
+    computerPickElem = document.getElementById('js-computerPick'),
+    playerResultElem = document.getElementById('js-playerResult'),
+    computerResultElem = document.getElementById('js-computerResult');
+
+function playerPick(playerPick) {
+    var computerPick = getComputerPick();
+    
+    playerPickElem.innerHTML = playerPick;
+    computerPickElem.innerHTML = computerPick;
+
+    checkRoundWinner(playerPick, computerPick);
+    setGamePoints();
+}    
+
+function checkRoundWinner(playerPick, computerPick) {
```

Comment:
Zauważ, że w tej funkcji robimy trochę masło maślane. Na podstawie picku gracza i komputera uzupełniamy zmienną `winnerIs`, a potem… sprawdzamy tą zmienną i zależnie od jej wartości podejmujemy akcje.

Spróbuj przerobić to tak żebyśmy zamiast wypełniać zmienną `winnerIs` - od razu podejmowali odpowiednie akcje. Te akcje najlepiej ubrać w nową funkcję bo w obecym stanie rzeczy ona robi zbyt wiele rzeczy na raz. Teoretycznie funkcja powinna robić maksymalnie jedną rzecz (zasada single responsibility)